### PR TITLE
Unary Ops Grad: Adds gradient for math.neg, math.exp and math.log

### PR DESCRIPTION
**Pull request by ManrajGrover**
This PR adds gradients for `math.neg`, `math.exp` and `math.log`.

---
This change is [Reviewable](https://reviewable.io/reviews/pair-code/deeplearnjs/588)

### CODE REVIEW

File: src/math/unary_ops_test.ts [hunk 1]
```diff
@@ -488,6 +558,47 @@ import {Array1D, Array2D, Scalar} from './ndarray';
...
+
+    it('gradients: Scalar', math => {
+      const a = Scalar.new(5);
+      const dy = Scalar.new(3);
+
+      const gradients = math.vjp(() => math.exp(a), a, dy);
+
+      expect(gradients.shape).toEqual(a.shape);
+      expect(gradients.dtype).toEqual('float32');
+      test_util.expectArraysClose(gradients, [3 * Math.exp(5)], 1);
```
Comment (ManrajGrover) [inline src/math/unary_ops_test.ts]: @nsthorat Here gradient value and expected value have a difference of more than `0.1`. Not sure why.

**Comment by dsmilkov** [comment]
Nice. Couple small comments. Keep up the great work!

---

Reviewed 1 of 2 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[src/math/unary_ops.ts, line 33 at r2](https://reviewable.io:443/reviews/pair-code/deeplearnjs/588#-L3muZMeCPX7CpUBk0bq:-L3muZMg4DFkZSV3S8sF:b-euntyy) ([raw file](https://github.com/pair-code/deeplearnjs/blob/ffc170e2789ac82c323090d363bbd8f932f70925/src/math/unary_ops.ts#L33)):*
> dy.mul(Scalar.new(-1))

shorter: `dy.neg();`

---

*[src/math/unary_ops.ts, line 66 at r2](https://reviewable.io:443/reviews/pair-code/deeplearnjs/588#-L3muknBqkN9HOkzefzl:-L3muknBqkN9HOkzefzm:b3b2ond) ([raw file](https://github.com/pair-code/deeplearnjs/blob/ffc170e2789ac82c323090d363bbd8f932f70925/src/math/unary_ops.ts#L66)):*
> ```XML
>     return ENV.engine.executeKernel(
>                'Exp', {inputs: {x}}, (dy: NDArray<R>, y: NDArray<R>) => {
>                  return {x: () => dy.mul(Ops.exp(x.asType('float32')))};
> ```

you can do: `dy.mul(y);` since y=exp(x) is already computed from the forward pass

---

*[src/math/unary_ops.ts, line 78 at r2](https://reviewable.io:443/reviews/pair-code/deeplearnjs/588#-L3mvLPcOJplVHxu7UD9:-L3mvLPcOJplVHxu7UDA:bzc4plw) ([raw file](https://github.com/pair-code/deeplearnjs/blob/ffc170e2789ac82c323090d363bbd8f932f70925/src/math/unary_ops.ts#L78)):*
> dy.div(x.asType('float32'))

we added `toFloat()`, easier to type than a string, so you can do: `dy.div(x.toFloat())`

---

*[src/math/unary_ops_test.ts, line 570 at r1](https://reviewable.io:443/reviews/pair-code/deeplearnjs/588#-L3gVTJ5SFrz0VYN4i6I-r1-570:-L3mvffJaGJTmqEV6E8k:bezjjxr) ([raw file](https://github.com/pair-code/deeplearnjs/blob/7c202696478fccaa9cfabb3ef19c58eda31a74b1/src/math/unary_ops_test.ts#L570)):*
<details><summary><i>Previously, ManrajGrover (Manraj Singh) wrote…</i></summary><blockquote>

@nsthorat Here gradient value and expected value have a difference of more than `0.1`. Not sure why.
</blockquote></details>

Is this happening only on the byte packing version of the test (where FLOAT_TEXTURE_ENABLED is false)? If yes, then try lowering x, maybe use 0.5 instead of 5. Fixing the byte packing version is high on our priority list.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pair-code/deeplearnjs/588#-:-L3mwIbn4otq4pvdC7Z1:byaekhl)*

**Comment by ManrajGrover** [comment]
@dsmilkov Yeps, it is happening where `FLOAT_TEXTURE_ENABLED` is false. I've faced this issue in another PR too.

Made the required changes.

**Comment by dsmilkov** [comment]
:lgtm_strong: Nice!

---

Review status: 0 of 2 files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pair-code/deeplearnjs/588#-:-L3oKzOcy2loV8Qy0SAu:b-pfdznz)*

**Pull request merged**